src/pages/NewProposal.tsx

```diff
@@ -15,780 +15,998 @@ import {
-              let callDataFunctionSignature = "0x0";
-              let callDataFunctionParamsEncoded = "";
-              
-              if (call.functionName.length === 0) {
-                callDataFunctionSignature = "0x0";
-              } else {
-                callDataFunctionSignature = library.eth.abi.encodeFunctionSignature(call.functionName)
-              }
-              
-              if (call.dataValues.length > 0) {
-                call.functionParams.map((functionParam, i) => {
-                  if (functionParam.type.indexOf("[]") > 0) {
-                    call.dataValues[i] = call.dataValues[i].slice(1,-1).split(',');
-                  }
-                });
-                callDataFunctionParamsEncoded = call.functionParams.length > 0 ? library.eth.abi.encodeParameters(
-                    call.functionParams.map((functionParam) => functionParam.type),
-                    call.dataValues
-                  ).substring(2)
-                  : "";
-              }
-              callData = callDataFunctionSignature + callDataFunctionParamsEncoded;
+  };
+
+  const createProposal = async function() {
+    console.debug('[RAW PROPOSAL]', titleText, ipfsHash, calls);
+    setSubmitionState(3);
+
+    const { library } = providerStore.getActiveWeb3React();
+
+    let to = [],
+      data = [],
+      value = [];
+    try {
+      if (schemeToUse.type !== 'ContributionReward') {
+        const callToController =
+          schemeToUse.controllerAddress === networkContracts.controller;
+
+        to = calls.map(call => {
+          return callToController ? networkContracts.controller : call.to;
+        });
+
+        data = calls.map(call => {
+          if (call.to === '') return '';
+
+          let callData;
+
+          if (call.callType === 'simple') {
+            let callDataFunctionSignature = '0x0';
+            let callDataFunctionParamsEncoded = '';
+
+            if (call.functionName.length === 0) {
+              callDataFunctionSignature = '0x0';
             } else {
-              callData = call.dataValues[0];
+              callDataFunctionSignature = library.eth.abi.encodeFunctionSignature(
+                call.functionName
+              );
             }
-            if (callToController && call.to !== networkContracts.controller) {
-              return daoService.encodeControllerGenericCall(
-                call.to,
-                callData,
-                call.callType === "simple" ? library.utils.toWei(call.value).toString()
-                : call.value
-              )
-            } else {
-              return callData;
+
+            if (call.dataValues.length > 0) {
+              call.functionParams.map((functionParam, i) => {
+                if (functionParam.type.indexOf('[]') > 0) {
+                  call.dataValues[i] = call.dataValues[i]
+                    .slice(1, -1)
+                    .split(',');
+                }
+                if (functionParam.type === 'bool') {
```

Comment:
This is the only piece of code you added?

Reply:
Yes